sylvia/tests/querier.rs

```diff
@@ -0,0 +1,131 @@
+use cosmwasm_std::{Addr, Response, StdResult};
+use cw_storage_plus::Item;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use sylvia::contract;
+use sylvia::types::InstantiateCtx;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, JsonSchema, Debug, Default)]
+pub struct CountResponse {
+    pub count: u64,
+}
+
+pub mod counter {
+    use cosmwasm_std::{Response, StdError, StdResult};
+    use sylvia::types::{ExecCtx, QueryCtx};
+    use sylvia::{contract, interface};
+
+    use crate::CountResponse;
+
+    #[interface]
+    pub trait Counter {
+        type Error: From<StdError>;
+
+        #[msg(query)]
+        fn count(&self, ctx: QueryCtx) -> StdResult<CountResponse>;
+
+        #[msg(exec)]
+        fn copy_count(&self, ctx: ExecCtx) -> StdResult<Response>;
+
+        #[msg(exec)]
+        fn set_count(&self, ctx: ExecCtx, new_count: u64) -> StdResult<Response>;
+    }
+
+    #[contract]
+    impl Counter for super::CounterContract<'_> {
+        type Error = StdError;
+
+        #[msg(query)]
+        fn count(&self, ctx: QueryCtx) -> StdResult<CountResponse> {
+            let count = self.count.load(ctx.deps.storage)?;
+            Ok(CountResponse { count })
+        }
+
+        #[msg(exec)]
+        fn set_count(&self, ctx: ExecCtx, new_count: u64) -> StdResult<Response> {
+            self.count.save(ctx.deps.storage, &new_count)?;
+            Ok(Response::new())
+        }
+
+        #[msg(exec)]
+        fn copy_count(&self, ctx: ExecCtx) -> StdResult<Response> {
+            let remote = self.remote.load(ctx.deps.storage)?;
+            let remote = Remote::from(&remote);
+            let other_count = remote.querier(&ctx.deps.querier).count()?.count;
+            self.count.save(ctx.deps.storage, &other_count)?;
+            Ok(Response::new())
+        }
+    }
+}
+
+pub struct CounterContract<'a> {
+    pub(crate) count: Item<'static, u64>,
+    pub(crate) remote: Item<'static, Remote<'a>>,
```

Comment:
This is unnecessary. Tests are not exposed from the library anyway, so if you want to make those fields public, just make them fully public. This `pub(crate)` immediately makes me look for a case when the `CounterContract` type leaks from the library which would be bad itself.